src/internal/connector/onedrive/item_handler.go

```diff
@@ -66,6 +70,72 @@ func (h itemBackupHandler) AugmentItemInfo(
 	item models.DriveItemable,
 	size int64,
 	parentPath *path.Builder,
+) details.ItemInfo {
+	return augmentItemInfo(dii, item, size, parentPath)
+}
+
+func (h itemBackupHandler) FormatDisplayPath(
+	_ string, // drive name not displayed for onedrive
+	pb *path.Builder,
+) string {
+	return "/" + pb.String()
+}
+
+func (h itemBackupHandler) NewLocationIDer(
+	driveID string,
+	elems ...string,
+) details.LocationIDer {
+	return details.NewOneDriveLocationIDer(driveID, elems...)
+}
+
+func (h itemBackupHandler) Requester() graph.Requester            { return h.ac.Requester }
+func (h itemBackupHandler) PermissionGetter() GetItemPermissioner { return h.ac }
+func (h itemBackupHandler) ItemGetter() GetItemer                 { return h.ac }
+
+// ---------------------------------------------------------------------------
+// Restore
+// ---------------------------------------------------------------------------
+
+var _ RestoreHandler = &itemRestoreHandler{}
+
+type itemRestoreHandler struct {
+	ac api.Drives
+}
+
+func NewRestoreHandler(ac api.Client) *itemRestoreHandler {
+	return &itemRestoreHandler{ac.Drives()}
+}
+
+// AugmentItemInfo will populate a details.OneDriveInfo struct
+// with properties from the drive item.  ItemSize is specified
+// separately for restore processes because the local itemable
+// doesn't have its size value updated as a side effect of creation,
+// and kiota drops any SetSize update.
+func (h itemRestoreHandler) AugmentItemInfo(
+	dii details.ItemInfo,
+	item models.DriveItemable,
+	size int64,
+	parentPath *path.Builder,
+) details.ItemInfo {
+	return augmentItemInfo(dii, item, size, parentPath)
+}
+
+func (h itemRestoreHandler) FolderByNameGetter() GetFolderByNamer          { return h.ac }
```

Comment:
Curious what are the benefits here of having 6 functions return the same data - `h.ac` ?

Reply:
I initially thought it would work as a way for tests to piecemeal their mock structures.  IE: you get to compile 6 different small mocks as opposed to one big one.  More modular, easily swappable, etc.

In practice, in later PRs, I found the implementation wasn't as elegant as the concept.  In fact, it caused more work than it solved.  So most/all of these have been transitioned to api client func wrappers instead.

Reply:
yeah, that would be mock combo hell. Thanks